File: src/main/java/me/nepnep/nepbot/message/command/commands/admin/RemoveBottomWhiteList.java
package me.nepnep.nepbot.message.command.commands.admin;

import me.nepnep.nepbot.database.WhiteListDatabase;
import me.nepnep.nepbot.message.command.Category;
import me.nepnep.nepbot.message.command.ICommand;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class RemoveBottomWhiteList implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {
        WhiteListDatabase.removeFromWhiteList(event.getChannel().getIdLong());
    }

    @Override
    public String getInvoke() {
        return "removebottomwhitelist";
    }

    @Override
    public Permission requiredPermission() {
        return Permission.MANAGE_CHANNEL;
    }

    @Override
    public Category getCategory() {
        return Category.ADMIN;
    }

    @Override
    public String getDescription() {
        return "Removes the channel from the bottom whitelist";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/admin/RemoveBottomWhitelist.kt
package me.nepnep.nepbot.message.command.commands.admin

import me.nepnep.nepbot.database.removeFromWhitelist
import me.nepnep.nepbot.message.command.Category
import me.nepnep.nepbot.message.command.AbstractCommand
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.entities.channel.middleman.GuildMessageChannel
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class RemoveBottomWhitelist : AbstractCommand(
    "removebottomwhitelist",
    Category.ADMIN,
    "Removes the channel from the bottom whitelist",
    Permission.MANAGE_CHANNEL
) {
    override suspend fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        channel.removeFromWhitelist()
    }
}